### ISSUE TITLE
Can I turn off GetLinkMetaData, it seriously affects the page loading speed.

### ISSUE THREAD

**Issue by yihanglee-cn** [usememos/memos#3255]
### Describe the bug

Can I turn off GetLinkMetaData, it seriously affects the page loading speed.

### Steps to reproduce

Can I turn off GetLinkMetaData, it seriously affects the page loading speed.
I don't need this feature at all! When I browse records with pictures, the page freezes very badly!
When I looked at the network data, I found that GetLinkMetaData was requested many times and the response time was very long！
![PixPin_2024-04-21_12-35-11](https://github.com/usememos/memos/assets/31716253/468ff770-cbaa-4b06-ae57-e22290225ff4)

### The version of Memos you're using

0.21

**Comment by MehadND** [comment]
I think this is necessary to give user the option to toggle this feature based on their needs. Since I mainly worked on the preview card feature. I will look into this.

**Comment by MehadND** [comment]
@boojack is this something that should be done? Allow some sort of toggle for Preview Card feature

**Comment by Jacob-zh** [comment]
I think so, 
Now, I putting the link in a block of code, this reduces HTTP requests.

**Comment by yihanglee-cn** [comment]
> > I think so, Now, I putting the link in a block of code, this reduces HTTP requests.
> 
> @Jacob-zh if you mean this way, then ![image](https://private-user-images.githubusercontent.com/34424878/324764445-28739c44-a8e7-48df-be32-393cc3d81881.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM4NjMwNzIsIm5iZiI6MTcxMzg2Mjc3MiwicGF0aCI6Ii8zNDQyNDg3OC8zMjQ3NjQ0NDUtMjg3MzljNDQtYThlNy00OGRmLWJlMzItMzkzY2MzZDgxODgxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDIzVDA4NTkzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYyYjkwMDZhODE0ZDVhOWE5MjA5MDAxZjdlYjNmMTE4OTEwYjlhMTNiNWJlZGE5MDdiMjY2OTBkZDM0NWI0Y2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Xu8euQG713G1NyGf8hoSR0staiDzYEjmp-lnmkxXv5I)
> 
> This reduces the HTTP request because the link is no longer considered a link anymore it's just a piece of text or line of code.
> 
> Therefore there's no request made to the Metadata endpoint.
> 
> If this is not what you mean then please do explain and help me understand

You're right, we'll make the links plain text, because too many requests will affects page performance and I don't even need link preview feature.

**Comment by MehadND** [comment]
I talked to the project owner and he feels that rather than having a switch to toggle it on/off. A better solution is to fix/optimise the Metadata service.

I already came up with some ideas like caching metadata for re-usability. 

This would still make a lot of requests at first depending on the links present in memos. But on second time or going to different pages it would check to see if there's request already exist in which case it would return the metadata stored rather than make a HTTP request.

Looks a lot easy to say but can be challenging to implement. But I will see what I can do.
